core/sdl/dreampicoport.cpp

```diff
@@ -110,11 +135,14 @@ class DreamPicoPortSerialHandler
 		}
 
 		asio::error_code ec;
-		serial_handler.open(serial_device, ec);
+		if (!serial_device.empty()) {
+			serial_handler.open(serial_device, ec);
+		}
 
 		if (ec || !serial_handler.is_open()) {
 			WARN_LOG(INPUT, "DreamPicoPort serial connection failed: %s", ec.message().c_str());
 			disconnect();
+			return;
 		} else {
 			NOTICE_LOG(INPUT, "DreamPicoPort serial connection successful!");
 		}
```

Comment:
nit: `else` is not needed, when the associated `if` body always returns.
```suggestion
}

NOTICE_LOG(INPUT, "DreamPicoPort serial connection successful!");
```